In [1]:
import pandas as pd
import numpy as np  
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

In [2]:
gdp_df = pd.read_csv('../data/gdp_per_capita.csv', skipfooter=2)

<ipython-input-2-ef274f2e1c61>:1: ParserWarning: Falling back to the 'python' engine because the 'c' engine does not support skipfooter; you can avoid this warning by specifying engine='python'.
  gdp_df = pd.read_csv('../data/gdp_per_capita.csv', skipfooter=2)


5. Added skipfooter = 2 to avoid the last two rows of the datafram which had some footer data pertaining to the data exluded for South Sudan

In [3]:
gdp_df.head(10)

Country or Area  Year        Value  Value Footnotes
0     Afghanistan  2019  2065.036398              NaN
1     Afghanistan  2018  2033.779002              NaN
2     Afghanistan  2017  2058.383832              NaN
3     Afghanistan  2016  2057.062164              NaN
4     Afghanistan  2015  2068.265904              NaN
5     Afghanistan  2014  2102.385234              NaN
6     Afghanistan  2013  2116.465454              NaN
7     Afghanistan  2012  2075.491747              NaN
8     Afghanistan  2011  1904.559799              NaN
9     Afghanistan  2010  1957.029338              NaN

In [4]:
gdp_df[-5:]

Country or Area  Year        Value  Value Footnotes
6863        Zimbabwe  1994  3263.934978              NaN
6864        Zimbabwe  1993  3033.504852              NaN
6865        Zimbabwe  1992  3054.889178              NaN
6866        Zimbabwe  1991  3426.598094              NaN
6867        Zimbabwe  1990  3324.348171              NaN

In [7]:
gdp_df = gdp_df.drop(columns = "Value Footnotes")

In [13]:
gdp_df = gdp_df.rename(columns={"Country or Area": "Country", "Value": "GDP_per_Capita"})

7. Rows = 3, Columns 6867

Country is an object, Year is a interger, GDP_per_Capita is a float

In [14]:
gdp_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 6868 entries, 0 to 6867
Data columns (total 3 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   Country         6868 non-null   object 
 1   Year            6868 non-null   int64  
 2   GDP_per_Capita  6868 non-null   float64
dtypes: float64(1), int64(1), object(1)
memory usage: 161.1+ KB


In [20]:
print(gdp_df["Year"].min())

print(gdp_df["Year"].max())

1990
2019


In [26]:
gdp_df["Year"].value_counts(sort = True)

2013    238
2014    238
2016    238
2015    238
2017    238
2012    237
2018    237
2011    237
2009    236
2010    236
2007    235
2008    235
2006    235
2004    234
2005    234
2002    233
2003    233
2001    232
2019    231
2000    231
1999    224
1998    223
1997    223
1996    221
1995    221
1994    214
1993    212
1992    209
1991    208
1990    207
Name: Year, dtype: int64

8. the number of observations increases from 1990 with 207 observations  until 2017 with 238. 2018 and 2019 decreased slightly with 237 and 231 observations, respectively. 

In [30]:
countries_of_gdp_df = gdp_df.drop_duplicates(subset="Country")
print(countries_of_gdp_df)

                  Country  Year  GDP_per_Capita
0             Afghanistan  2019     2065.036398
18                Albania  2019    13724.058790
48                Algeria  2019    11510.557088
78                 Angola  2019     6670.332297
108   Antigua and Barbuda  2019    21910.184985
...                   ...   ...             ...
6722              Vietnam  2019     8041.178551
6752   West Bank and Gaza  2019     6219.960291
6778                World  2019    16911.193295
6808               Zambia  2019     3470.448801
6838             Zimbabwe  2019     2835.948532

[238 rows x 3 columns]


9. 238 Countries are in the data set.

In [32]:
gdp_df["Country"].value_counts(sort = True)

Jordan                        30
Early-demographic dividend    30
East Asia & Pacific           30
Malaysia                      30
Low income                    30
                              ..
Nauru                         16
Cayman Islands                13
Sint Maarten (Dutch part)     10
Turks and Caicos Islands       9
Djibouti                       7
Name: Country, Length: 238, dtype: int64